chronicle/src/main/java/net/openhft/chronicle/tcp/SinkTcp.java

```diff
@@ -65,7 +84,8 @@ public boolean open(boolean blocking) throws IOException {
                 socketChannel.socket().setSendBufferSize(this.builder.sendBufferSize());
             }
 
-            super.setSocketChannel(socketChannel);
+            this.socketChannel = socketChannel;
+            this.tcpEventHandler = new TcpEventHandler(socketChannel, builder.tcpPipeline(sinkTcpHandler), sessionDetailsProvider, builder.sendBufferSize(), builder.receiveBufferSize());
```

Comment:
Yeah, sure.  I'll take a look at it in the morning but I think it should be achievable on reads.  I'll see if it's easy enough to provide blocking on the write as well.